File: Aruvi21042026/Aruvi21042026/Arivi App Final/app/src/main/java/com/Saalai/SalaiMusicApp/Models/TopNavItem.java
package com.Saalai.SalaiMusicApp.Models;

import com.google.gson.annotations.SerializedName;

public class TopNavItem {
    @SerializedName("topmenuId")
    private int topmenuId;

    @SerializedName("topmenuName")
    private String topmenuName;

    @SerializedName("topmenuStatusId")
    private int topmenuStatusId;

    @SerializedName("topmenuStatus")
    private String topmenuStatus;

    @SerializedName("topmenuActiveIcon")
    private String topmenuActiveIcon;

    @SerializedName("topmenuInActiveIcon")
    private String topmenuInActiveIcon;

    @SerializedName("is_active") // If this field exists in your API
    private boolean isActive;

    // Getters and Setters
    public int getTopmenuId() {
        return topmenuId;
    }

    public void setTopmenuId(int topmenuId) {
        this.topmenuId = topmenuId;
    }

    public String getTopmenuName() {
        return topmenuName;
    }

    public void setTopmenuName(String topmenuName) {
        this.topmenuName = topmenuName;
    }

    public int getTopmenuStatusId() {
        return topmenuStatusId;
    }

    public void setTopmenuStatusId(int topmenuStatusId) {
        this.topmenuStatusId = topmenuStatusId;
    }

    public String getTopmenuStatus() {
        return topmenuStatus;
    }

    public void setTopmenuStatus(String topmenuStatus) {
        this.topmenuStatus = topmenuStatus;
    }

    public String getTopmenuActiveIcon() {
        return topmenuActiveIcon;
    }

    public void setTopmenuActiveIcon(String topmenuActiveIcon) {
        this.topmenuActiveIcon = topmenuActiveIcon;
    }

    public String getTopmenuInActiveIcon() {
        return topmenuInActiveIcon;
    }

    public void setTopmenuInActiveIcon(String topmenuInActiveIcon) {
        this.topmenuInActiveIcon = topmenuInActiveIcon;
    }

    public boolean isActive() {
        return isActive;
    }

    public void setActive(boolean active) {
        isActive = active;
    }
}

File: aruviflutter/lib/models/top_nav_item.dart
class TopNavItem {
  final int topmenuId;
  final String topmenuName;
  final int? topmenuStatusId;
  final String? topmenuStatus;
  final String? topmenuActiveIcon;
  final String? topmenuInActiveIcon;
  final bool? isActive;

  TopNavItem({
    required this.topmenuId,
    required this.topmenuName,
    this.topmenuStatusId,
    this.topmenuStatus,
    this.topmenuActiveIcon,
    this.topmenuInActiveIcon,
    this.isActive,
  });

  factory TopNavItem.fromJson(Map<String, dynamic> json) {
    return TopNavItem(
      topmenuId: json['topmenuId'] is int ? json['topmenuId'] : int.tryParse(json['topmenuId']?.toString() ?? '0') ?? 0,
      topmenuName: json['topmenuName']?.toString() ?? '',
      topmenuStatusId: json['topmenuStatusId'] is int ? json['topmenuStatusId'] : int.tryParse(json['topmenuStatusId']?.toString() ?? ''),
      topmenuStatus: json['topmenuStatus']?.toString(),
      topmenuActiveIcon: json['topmenuActiveIcon']?.toString(),
      topmenuInActiveIcon: json['topmenuInActiveIcon']?.toString(),
      isActive: json['is_active'] is bool ? json['is_active'] : json['is_active']?.toString().toLowerCase() == 'true',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'topmenuId': topmenuId,
      'topmenuName': topmenuName,
      'topmenuStatusId': topmenuStatusId,
      'topmenuStatus': topmenuStatus,
      'topmenuActiveIcon': topmenuActiveIcon,
      'topmenuInActiveIcon': topmenuInActiveIcon,
      'is_active': isActive,
    };
  }
}
